### DISCUSSION TITLE
Multiple Jaas authentications

### DISCUSSION THREAD

**Discussion by GuihToledo** [spring-projects/spring-kafka#2031] [Q&A] [answered]
Hello ! 

I can't find a way to configure multiple JaasConfigurations (using KafkaJaasLoginModuleInitializer) to connect in different kafka clusters, currently we can create different clusters connection, but they share the same authentication. 

Would be an fine option to set which authentication  should be used for each DefaultKafkaConsumerFactory/DefaultKafkaProducerFactory.

**Comment by garyrussell** [comment]
Issues are not for asking questions - converting to a discussion.

**Comment by garyrussell** [comment, marked as answer]
It can't be done using `KafkaJaasLoginModuleInitializer` - it sets the JVM global JAAS config.

You have to configure it via producer/consumer configs; e.g. https://kafka.apache.org/documentation/#consumerconfigs_sasl.jaas.config

**Reply by GuihToledo** [reply]
Thank you for your support ! Worked fine.
